openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/FileLayerProvider.scala

```diff
@@ -59,15 +64,49 @@ class BandCompositeRasterSource(override val sources: NonEmptyList[RasterSource]
     }
 
   }
-  override def cellType: CellType = sources.map(_.cellType).reduceLeft(_ union _)
+
+  override def cellType: CellType = {
+    var _cellType = sources.map(_.cellType).reduceLeft(_ union _)
+    if (pixelValueOffset < 0) {
+      // Big chance the value will go under 0, so type needs to be signed
+      _cellType = _cellType match {
+        case UByteCellType => ByteCellType
+        case UByteConstantNoDataCellType => ByteConstantNoDataCellType
+        case UShortCellType => ShortCellType
+        case UShortConstantNoDataCellType => ShortConstantNoDataCellType
+        case default => default // Was probably already signed
+      }
+    }
+    _cellType
+  }
 
   override def name: SourceName = sources.head.name
   override def bandCount: Int = sources.size
 
+  private def offsetBand(band: Tile): Tile = {
+    if (pixelValueOffset == 0) {
+      band
+    } else {
+      band match {
+        case b: geotrellis.raster.UShortArrayTile =>
+          geotrellis.raster.ShortConstantNoDataArrayTile(b.array.map(x =>
+            if (x == ushortNODATA) shortNODATA else (x + pixelValueOffset).toShort), band.cols, band.rows)
+        case b: geotrellis.raster.UByteArrayTile =>
+          geotrellis.raster.ByteConstantNoDataArrayTile(b.array.map(x =>
+            if (x == ubyteNODATA) byteNODATA else (x + pixelValueOffset).toByte), band.cols, band.rows)
+        case _ =>
+          // Not sure how to handle user defined nodata for example
+          throw new IllegalArgumentException("Can not yet combine 'pixelValueOffset' and '" + band.getClass.getName + "'. See more: https://github.com/Open-EO/openeo-geotrellis-extensions/issues/144")
```

Comment:
@bossie , ok to refer to ticket here?

Reply:
True, I forget to label some commits. I'll see if I can put a pre-commit hook to remind me